dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/main/java/org/apache/dolphinscheduler/plugin/task/api/utils/K8sUtils.java

```diff
@@ -96,7 +96,7 @@ public String getPodLog(String jobName, String namespace) {
             String podName = null;
             for (Pod pod : podList) {
                 podName = pod.getMetadata().getName();
-                if (jobName.equals(podName.substring(0, pod.getMetadata().getName().lastIndexOf("-")))) {
+                if (podName.contains("-") && jobName.equals(podName.substring(0, podName.lastIndexOf("-")))) {
```

Comment:
hi @Radeity because some pods may also run in the same namespace as the pods created by DS. Therefore, the naming rules for pods may not necessarily be the same as those generated by DS.